src/utils/saml.ts

```diff
@@ -1286,7 +1305,38 @@ export const getPreValidateResponse = (
         )
       ).map(() => _)
     )
-    .bimap(callback, _ => callback(null, true, _.InResponseTo))
+    .bimap(
+      error => {
+        if (telemetryClient) {
+          telemetryClient.trackEvent({
+            name: "backend.login.prevalidate",
+            properties: {
+              message: error.message,
+              type: "ERROR"
+            }
+          });
+        }
+        return callback(error);
+      },
+      _ => {
+        // Check if Response Signature is missing
+        const signatureOfResponseCount = (xpath.select(
+          "count(//*[local-name(.)='Response']/*[local-name(.)='Signature'])",
+          doc
```

Comment:
can we use DOM parser and remove xpath dependency instead?

Reply:
Yes, but the code will be a little more complex
```typescript
const signatureOfResponseCount =
  _.Response.getElementsByTagNameNS(SAML_NAMESPACE.XMLDSIG, "Signature")
    .length -
  _.Assertion.getElementsByTagNameNS(
    SAML_NAMESPACE.XMLDSIG,
    "Signature"
  ).length;
```

Reply:
yes that's right, we must chose one between the two approaches and try to not mix them. 
(if I'd start again I'd rather have used https://www.npmjs.com/package/simple-xml instead of xpath / dom :-)

Reply:
Fixed in a729df3